runtime/compiler/z/codegen/J9S390CHelperLinkage.cpp

```diff
@@ -364,6 +361,17 @@ TR::Register * TR::S390CHelperLinkage::buildDirectDispatch(TR::Node * callNode,
       }
    TR::SymbolReference * callSymRef = callNode->getSymbolReference();
    void * destAddr = callNode->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethodAddress();
+
+#if defined(J9ZOS390)
+   // TR_S390numRuntimeHelpers represents invalid helper index
+   if (helperIndex != TR_S390numRuntimeHelpers)
```

Comment:
@NigelYiboYu I see that a declaration of this function defines `helperIndex` with `TR_S390numRuntimeHelpers`  and for JNI linkage calls (for pre/post offload check) , this function `(buildDirectDispatch)` will be called  with `TR_RuntimeHelper` equal to `TR_S390jitPreJNICallOffloadCheck` or `TR_S390jitPostJNICallOffloadCheck`. Is it correct?

Reply:
Yes, JNI IFA uses TR_S390jitPreJNICallOffloadCheck and TR_S390jitPostJNICallOffloadCheck.
For the lack of a better choice, I use `TR_S390numRuntimeHelpers` to validate the helper index parameter.

Reply:
Okk. I think better choice here would be use of this [isFastPathOnly](https://github.com/NigelYiboYu/openj9/blob/5ee8d1852d8bf8a8608e74bdded9a510737119b2/runtime/compiler/z/codegen/J9S390CHelperLinkage.cpp#L271). I think we only set this when we are calling a C helper function directly from JIT'd code. This should also cover the case for `instanceOf` and load environment register for that.